File: mobile/lib/models/routine.dart
class Routine {
  final int id;
  final int userId;
  final String wakeTime;
  final String breakfastTime;
  final String lunchTime;
  final String dinnerTime;
  final String sleepTime;

  Routine({
    required this.id,
    required this.userId,
    required this.wakeTime,
    required this.breakfastTime,
    required this.lunchTime,
    required this.dinnerTime,
    required this.sleepTime,
  });

  factory Routine.fromJson(Map<String, dynamic> json) {
    return Routine(
      id: json['id'],
      userId: json['user_id'],
      wakeTime: json['wake_time'] ?? '07:00',
      breakfastTime: json['breakfast_time'] ?? '08:00',
      lunchTime: json['lunch_time'] ?? '13:00',
      dinnerTime: json['dinner_time'] ?? '20:00',
      sleepTime: json['sleep_time'] ?? '23:00',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'wake_time': wakeTime,
      'breakfast_time': breakfastTime,
      'lunch_time': lunchTime,
      'dinner_time': dinnerTime,
      'sleep_time': sleepTime,
    };
  }
}

File: backend/schemas/routine.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class RoutineBase(BaseModel):
    wake_time: str = "07:00"
    breakfast_time: str = "08:00"
    lunch_time: str = "13:00"
    dinner_time: str = "20:00"
    sleep_time: str = "23:00"


class RoutineCreate(RoutineBase):
    pass


class RoutineUpdate(RoutineBase):
    wake_time: Optional[str] = None
    breakfast_time: Optional[str] = None
    lunch_time: Optional[str] = None
    dinner_time: Optional[str] = None
    sleep_time: Optional[str] = None


class Routine(RoutineBase):
    id: int
    user_id: int

    class Config:
        from_attributes = True
